core/server/worker/src/main/java/alluxio/worker/block/evictor/LRFUEvictor.java

```diff
@@ -78,9 +79,9 @@ public LRFUEvictor(BlockMetadataManagerView view, Allocator allocator) {
         "Attenuation factor should be no less than 2.0");
 
     // Preloading blocks
-    for (StorageTierView tier : mManagerView.getTierViews()) {
+    for (StorageTierView tier : mMetadataView.getTierViews()) {
       for (StorageDirView dir : tier.getDirViews()) {
-        for (BlockMeta block : dir.getEvictableBlocks()) {
+        for (BlockMeta block : ((StorageDirEvictorView) dir).getEvictableBlocks()) {
```

Comment:
`StorageDirView` is casted to `StorageDirEvictorView` at many places. Can we avoid that somehow? Maybe `StorageTierEvictorView` could implement a one that returns `StorageDirEvictorView` instead.

Reply:
I tried to use generic view of `StorageTierView` and `StorageDirView` and implement the abstract methods in allocator and evictor classes. The detail code listed in https://github.com/LuQQiu/alluxio/pull/1. This way reduce many unneeded cast.

However, in the allocators, we directly operate on abstract class `BlockMetadataView`.
And intellij will complain:
```
for (StorageTierView tierView : mMetadataView.getTierViews()) {
        for (StorageDirView dirView : tierView.getDirViews()) { <-- complain about incompatible types
          if (dirView.getMediumType().equals(mediumType)
```
Incompatible types: required object, but found StorageDirView